controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -184,6 +172,15 @@ func CreateReconciliationPlan(instance *dsc.DataScienceCluster) *profiles.Reconc
 	profiles.ProfileConfigs = profiles.SetDefaultProfiles()
 	// Create plan for component deployment
 	profiles.PopulatePlan(profiles.ProfileConfigs[profile], plan, instance)
-	// Similarly set other profiles
+
+	// warn of odd profile combinations
+	if profile == profiles.ProfileServing && !plan.Components[modelmeshserving.ComponentName] {
+		r.Log.Info("warning: profile '%v' has been selected, but component %s has been explicitly disabled in the CR", profile, modelmeshserving.ComponentName)
+	} else if profile == profiles.ProfileTraining && !plan.Components[datasciencepipelines.ComponentName] {
+		r.Log.Info("warning: profile '%v' has been selected, but component %s has been explicitly disabled in the CR", profile, datasciencepipelines.ComponentName)
+	} else if profile == profiles.ProfileWorkbench && !plan.Components[workbenches.ComponentName] {
+		r.Log.Info("warning: profile '%v' has been selected, but component %s has been explicitly disabled in the CR", profile, workbenches.ComponentName)
+	}
```

Comment:
did some changes, to use switch-case for readability also easy to expand when we add more profile or components later
```suggestion
var componentNames []string

switch profile {
case profiles.ProfileServing:
	componentNames = []string{modelmeshserving.ComponentName}
case profiles.ProfileTraining:
	componentNames = []string{datasciencepipelines.ComponentName}
case profiles.ProfileWorkbench:
	componentNames = []string{workbenches.ComponentName}
case profiles.ProfileCore: // initialize with all core components
	componentNames = []string{
		modelmeshserving.ComponentName,
		datasciencepipelines.ComponentName,
		workbenches.ComponentName,
		dashboard.ComponentName,
	}
case profiles.ProfileNone:
	// skip logger if profile to none
default: // unsupported profile, should not fall here if we set enum for profile
	r.Log.warning("warning: unrecognized profile '%v' selected", profile)
}

for _, comp := range componentNames {
	if !plan.Components[comp] {
		r.Log.Info("warning: profile '%v' has been selected, but component %s has been explicitly disabled in the CR", profile, comp)
	}
}
```

Reply:
I don't think we should log any warning for the `core` profile as it will be common for someone to select the core profile and disable components. I will update the code to avoid the duplication of the log statement.